partner_preview_scripts/ValidationLab.py

```diff
@@ -3,432 +3,472 @@
+            "releaseDate",
+        ]
+
+        return (
+            processed_experiment.join(
+                parsed_assays, on=["targetFromSourceId", "cellLineName"], how="inner"
+            )
+            .select(
+                # Columns from the raw evidence file:
+                "targetFromSourceId",
+                "cellLineName",
+                "assessment",
+                "resourceScore",
+                "primaryProjectHit",
+                # Column from the assay parser:
+                "assays",
+                # Columns from the project metadata:
+                *[
+                    f.lit(getattr(self, colname)).alias(colname)
+                    for colname in columns_to_add
+                ],
+            )
+            .join(biomarkers, on="cellLineName", how="left")
         )
-        # Grouping data by cell lines again:
-        .groupBy("cellName")
-        .agg(
-            f.collect_list("biomarkers").alias("biomarkers"),
-            f.first(f.col("diseaseCellLines")).alias("diseaseCellLines"),
+
+    @staticmethod
+    def _process_raw_evidence(raw_data: DataFrame, assays: list) -> DataFrame:
+        # Read full evidence data:
+        return raw_data.select(
+            # Extract target name:
+            f.col("gene_ID").alias("targetFromSourceId"),
+            # Extract cell-line name:
+            f.col("cell_line_ID").alias("cellLineName"),
+            # Extract VL assessment:
+            f.regexp_replace(f.col("OTVL_Assessment"), r"\n", " ").alias("assessment"),
+            f.col("OTAR Primary Project Hit")
+            .cast(t.BooleanType())
+            .alias("primaryProjectHit"),
+            # Extract assessment score:
+            f.col("OTVL_Assessment_Score").cast(t.FloatType()).alias("resourceScore"),
+            # Extract all assays:
+            *[
+                f.col(assay["label"]).cast(t.BooleanType()).alias(assay["shortName"])
+                for assay in assays
+                if assay["label"] in raw_data.columns
+            ],
         )
-        .persist()
-    )
 
-    # Reading and processing hypothesis data:
-    hypothesis_generator = ParseHypotheses(spark)
-    validation_hypotheses_df = hypothesis_generator.parse_hypotheses(
-        expectedFile=hypothesis_expected_file, observedFile=hypothesis_observed_file
-    )
+    @staticmethod
+    def _parser_assay_object(
+        raw_evidence_df: DataFrame, assays: list, spark: SparkSession
+    ) -> DataFrame:
+        """Organise experimental data into the right shape."""
+        # Generate unpivot expression - not all assays are present in the data, some project might done with fewer assays:
+        assay_names = [
+            assay["shortName"]
+            for assay in assays
+            if assay["shortName"] in raw_evidence_df.columns
+        ]
+
+        unpivot_expression = f"""stack({len(assay_names)}, {', '.join([f"'{assay}', `{assay}`" for assay in assay_names])}) as (shortName, isHit)"""
```

Comment:
I think this is the same function as the unstacking one. I wonder the same, whether we can use the native `unpivot`

Reply:
Once we decided to bump pyspark version we can review the repo for other occurrences of this pattern.